daphne_worker/src/error_reporting.rs

```diff
@@ -0,0 +1,21 @@
+// Copyright (c) 2023 Cloudflare, Inc. All rights reserved.
+// SPDX-License-Identifier: BSD-3-Clause
+
+//! Daphne-Worker error reporting trait and default implementation.
+
+use daphne::aborts::DapAbort;
+
+/// Interface for error reporting in Daphne
+/// Refer to StderrErrorReporter for implementation example.
+pub trait ErrorReporter {
+    fn report_abort(&self, error: &DapAbort);
+}
+
+/// Default implementation of the error reporting trait. Displays error on stderr.
+pub(crate) struct StderrErrorReporter {}
+
+impl ErrorReporter for StderrErrorReporter {
+    fn report_abort(&self, error: &DapAbort) {
+        eprintln!("ABORT: {}", error);
```

Comment:
Hrm, is this the default behavior we want? I don't think `npx wrangler dev` will actually print to stderr. Perhaps we should use `tracing::error!()` instead?